src/librustc_resolve/lib.rs

```diff
@@ -2934,6 +2990,24 @@ impl<'a> Resolver<'a> {
     pub fn all_macros(&self) -> &FxHashMap<Symbol, Res> {
         &self.all_macros
     }
+
+    pub fn placeholder_field_index(&self, node_id: NodeId) -> usize {
+        self.placeholder_field_indices[&node_id]
+    }
+
+    pub fn set_placeholder_field_index(&mut self, node_id: NodeId, index: usize) {
+        let old_index = self.placeholder_field_indices.insert(node_id, index);
+        assert!(old_index.is_none(), "placeholder field index is reset for a node ID");
+    }
+
+    pub fn invocation_parent(&self, invoc_id: ExpnId) -> LocalDefId {
+        self.invocation_parents[&invoc_id]
+    }
+
+    pub fn set_invocation_parent(&mut self, invoc_id: ExpnId, parent: LocalDefId) {
+        let old_parent = self.invocation_parents.insert(invoc_id, parent);
+        assert!(old_parent.is_none(), "parent `LocalDefId` is reset for an invocation");
+    }
```

Comment:
These `(set_)(placeholder_field_index,invocation_parent)` getters and setters are useless now, they provide access to already visible `Resolver` fields and are used only once.

Reply:
Useless methods removed.